server/pbench/bin/pbench-server-prep-shim-001

```diff
@@ -0,0 +1,213 @@
+#! /bin/bash
+
+# set -x
+
+# This shim is used to prepare the tarballs that a version 001 client
+# submits for further processing. It copies the tarballs and their MD5
+# sums (plus an optional prefix file) to the archive (after checking)
+# and sets the state links, so that the dispatch script will pick them
+# up and get the ball rolling. IOW, it does impedance matching between
+# version 001 clients and the server scripts.
+
+# There is an enormous amount of code duplication between this script
+# and the version 002 script. TBD: refactor (but maybe wait and do it
+# for the inotify versions - the crontab version has a very limited life
+# span at this point).
+
+###########################################################################
+# load common things
+opts=$SHELLOPTS
+case $opts in
+    *xtrace*)
+        dir=$(dirname $(which $0))
+        PROG=$(basename $(which $0))
+        ;;
+    *)
+        dir=$(dirname $0)
+        PROG=$(basename $0)
+        ;;
+esac
+
+. $dir/pbench-base.sh
+###########################################################################
+
+archive=$(getconf.py archive-dir pbench-server)
+qdir=$(getconf.py quarantine-dir pbench-server)
+quarantine=${qdir}/md5-001
+duplicates=${qdir}/duplicates-001
+
+# The following directory holds tarballs that are quarantined because
+# of operational errors on the server. They should be retried after
+# the problem is fixed: basically, move them back into the reception
+# area for 001 agents and wait.
+errors=${qdir}/errors-001
+
+errlog=$LOGSDIR/$PROG/$PROG.error
+if [[ "$archive" == "" ]] ;then
+    echo "Failed: \"getconf.py archive-dir pbench-server" >> $errlog
+    exit 1
+fi
+
+if [[ "$quarantine" == "" ]] ;then
```

Comment:
Won't `$quarantine` never be an empty string?

Reply:
Ah, right, I should be checking $qdir. I'll fix.